dotnet/src/SemanticKernel.Abstractions/Filters/AutoFunctionInvocation/AutoFunctionInvocationContext.cs

```diff
@@ -62,6 +66,16 @@ public class AutoFunctionInvocationContext
     /// </summary>
     public int FunctionCount { get; init; }
 
+    /// <summary>
+    /// The ID of the tool call.
+    /// </summary>
+    public string? ToolCallId { get; init; }
```

Comment:
In parallel tool calls scenarios you might have multiple ToolIds per message.

See:  https://github.com/microsoft/semantic-kernel/blob/6044fbb2d50da7de54b4e5a5659863c22b7f6b35/dotnet/src/Connectors/Connectors.OpenAIV2/Core/OpenAIChatMessageContent.cs#L72

```suggestion
    public List<string>? ToolCallIds { get; init; }
```

Reply:
Thanks for your feedback! I think even in parallel scenario, the filter will be executed per tool/function invocation, so the context should have unique ID associated with specific tool. This will allow to control all auto function invocations separately.